packages/fiori/src/ViewSettingsDialogTemplate.tsx

```diff
@@ -75,74 +104,97 @@ function ViewSettingsDialogTemplateContent(this: ViewSettingsDialog) {
 			}}
 		>
 			{this.shouldBuildSort && this.isModeSort && (
-				<div class="ui5-vsd-sort">
-					<List
-						selectionMode="SingleStart"
-						onSelectionChange={this._onSortOrderChange} // radio button - use selectionChange
-						sort-order=""
-						accessibleNameRef={`${this._id}-label`}
-					>
-						<ListItemGroup headerText={this._sortOrderLabel}>
-							{this._currentSettings.sortOrder.map(item => (
-								<ListItemStandard
-									selected={item.selected}
-								>{item.text}</ListItemStandard>
-							))}
-						</ListItemGroup>
-					</List>
-					<List
-						selectionMode="SingleStart"
-						onSelectionChange={this._onSortByChange} // radio button - use selectionChange
-						sort-by=""
-					>
-						<ListItemGroup headerText={this._sortByLabel}>
-							{this._currentSettings.sortBy.map((item, index) => (
-								<ListItemStandard
-									data-ui5-external-action-item-index={index}
-									selected={item.selected}
-								>{item.text}</ListItemStandard>
-							))}
-						</ListItemGroup>
-					</List>
-				</div>
+				ViewSettingsDialogSortAndGroupTemplate.call(this, true)
+			)}
+
+			{this.shouldBuildFilter && this.isModeFilter && (
+				ViewSettingsDialogFilterTemplate.call(this)
 			)}
-			{this.shouldBuildFilter && this.isModeFilter && (<>
-				{this._filterStepTwo ? (
-					<List
-						selectionMode="Multiple"
-						onSelectionChange={this._handleFilterValueItemClick} // checkboxes to select/deselect - use selectionChange
-						accessibleNameRef={`${this._id}-label`}
-					>
-						{this._currentSettings.filters.filter(item => item.selected).map(item => (<>
-							{item.filterOptions.map(option => (
-								<ListItemStandard
-									selected={option.selected}
-								>{option.text}</ListItemStandard>
-							))}
-						</>))}
-					</List>
-				) : ( // else
-					<List
-						onItemClick={this._changeCurrentFilter} // list item to drill down into the second-level menu - use click
-						accessibleNameRef={`${this._id}-label`}
-					>
-						<ListItemGroup headerText={this._filterByLabel}>
-							{this.filterItems.map(item => (
-								<ListItemStandard
-									class="ui5-vsd-filterItemList"
-									// selected={item.selected} TODO
-									additionalText={item.additionalText}
-									accessibleName={this._selectedFiltersLabel(item)}
-								>{item.text}</ListItemStandard>
-							))}
-						</ListItemGroup>
-					</List>
-				)}
-			</>)}
+
+			{this.shouldBuildGroup && this.isModeGroup && (
+				ViewSettingsDialogSortAndGroupTemplate.call(this, false)
+			)}
+
 		</div>
 	);
 }
 
+function ViewSettingsDialogSortAndGroupTemplate(this: ViewSettingsDialog, sortMode: boolean) {
```

Comment:
sortMode does not sound good to me as mode can be confused with some sort of sort configuration. For more clarification i would rather go isSortTemplate